src/library/filescanner.c

```diff
@@ -1838,6 +1888,7 @@ playlist_add_files(FILE *fp, int pl_id, const char *virtual_path)
       return -1;
     }
 
+  added = 0;
```

Comment:
Instead of counting the number of inserts, you can directly query `qp.results`:

```
if (qp.results > 0)
  {
    while (((ret = db_query_fetch_file(&qp, &dbmfi)) == 0) && (dbmfi.id))
    ...
  }
else
  {
     // Handle items not in the files table
  }
```